presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
I think it's better just to use the `text` types which don't have an explicit length. That avoids all the problems with declared row size limits. Given that our current support is only for `CTAS`, it is very unlikely that users are trying to create a table with specific column sizes -- they just want to write the data to MySQL.

I'm not concerned about the size error being caught by analysis. There are plenty of data errors that can't be caught at analysis time.